itests/kit/node_unmanaged.go

```diff
@@ -440,66 +379,228 @@ func (tm *TestUnmanagedMiner) waitForMutableDeadline(ctx context.Context, sector
 	tm.currentSectorNum++
+	return sectorNumber
+}
+
+func (tm *TestUnmanagedMiner) PrepareSectorForProveCommit(
+	ctx context.Context,
+	proofType abi.RegisteredSealProof,
+	sectorNumber abi.SectorNumber,
+	pieces []abi.PieceInfo,
+) (seedEpoch abi.ChainEpoch, proveCommit []byte) {
+
+	req := require.New(tm.t)
+
+	// Wait for the pre-commitseal randomness to be available (we can only draw seal randomness from tipsets that have already achieved finality)
+	preCommitSealRandEpoch := tm.waitPreCommitSealRandomness(ctx, sectorNumber, proofType)
+
+	// Generate a Pre-Commit for the CC sector -> this persists the proof on the `TestUnmanagedMiner` Miner State
+	tm.generatePreCommit(ctx, sectorNumber, preCommitSealRandEpoch, proofType, pieces)
 
 	// --------------------Create pre-commit for the CC sector -> we'll just pre-commit `sector size` worth of 0s for this CC sector
 
-	// Step 1: Wait for the pre-commitseal randomness to be available (we can only draw seal randomness from tipsets that have already achieved finality)
-	preCommitSealRand := tm.waitPreCommitSealRandomness(ctx, sectorNumber)
+	if !proofType.IsNonInteractive() {
+		// Submit the Pre-Commit to the network
+		var uc *cid.Cid
+		if len(pieces) > 0 {
+			unsealedCid := tm.unsealedCids[sectorNumber]
+			uc = &unsealedCid
+		}
+		r, err := tm.SubmitMessage(ctx, &miner14.PreCommitSectorBatchParams2{
+			Sectors: []miner14.SectorPreCommitInfo{{
+				Expiration:    2880 * 300,
+				SectorNumber:  sectorNumber,
+				SealProof:     proofType,
+				SealedCID:     tm.sealedCids[sectorNumber],
+				SealRandEpoch: preCommitSealRandEpoch,
+				UnsealedCid:   uc,
+			}},
+		}, 1, builtin.MethodsMiner.PreCommitSectorBatch2)
+		req.NoError(err)
+		req.True(r.Receipt.ExitCode.IsSuccess())
+	}
+
+	// Generate a ProveCommit for the CC sector
+	var seedRandomness abi.InteractiveSealRandomness
+	seedEpoch, seedRandomness = tm.proveCommitWaitSeed(ctx, sectorNumber, proofType)
 
+	proveCommit = []byte{0xde, 0xad, 0xbe, 0xef} // mock prove commit
 	if !tm.mockProofs {
-		// Step 2: Write empty bytes that we want to seal i.e. create our CC sector
-		tm.makeAndSaveCCSector(ctx, sectorNumber)
+		proveCommit = tm.generateProveCommit(ctx, sectorNumber, proofType, seedRandomness, pieces)
+	}
+
+	return seedEpoch, proveCommit
+}
+
+func (tm *TestUnmanagedMiner) SubmitProveCommit(
+	ctx context.Context,
+	proofType abi.RegisteredSealProof,
+	sectorNumber abi.SectorNumber,
+	seedEpoch abi.ChainEpoch,
+	proveCommit []byte,
+	pieceManifest []miner14.PieceActivationManifest,
+) {
+
+	req := require.New(tm.t)
+
+	if proofType.IsNonInteractive() {
+		req.Nil(pieceManifest, "piece manifest should be nil for Non-interactive PoRep")
+	}
+
+	// Step 6: Submit the ProveCommit to the network
+	if proofType.IsNonInteractive() {
+		tm.t.Log("Submitting ProveCommitSector ...")
+
+		var provingDeadline uint64 = 7
+		if tm.IsUmmutableDeadline(ctx, provingDeadline) {
+			// avoid immutable deadlines
+			provingDeadline = 5
```

Comment:
What's this with hard coding the "5" and "7" deadlines here ?

Reply:
I just want a stable deadline to assert against when it lands (see below from here), but if it's immutable then it won't work and this test will be flaky, so I have to move it to another stable deadline. It can't be 6 because we might be in 6 (i.e. 6&7 are immutable), so I jump back to 5 which I know will be mutable since it's 2 epochs removed from now and we're unlikely to accidentally trip forward into an immutable one. I guess >9 would also be somewhat safe but <=5 seems like the safest option.